frontend/hooks/data-fetching/course.ts

```diff
@@ -336,6 +336,28 @@ export async function createQueue(
     }
 }
 
+export async function partialUpdateQueue(
+    courseId: number,
+    queueId: number,
+    payload: Partial<Queue>
+): Promise<void> {
+    const res = await doApiRequest(
+        `/api/courses/${courseId}/queues/${queueId}/`,
+        {
+            method: "PATCH",
+            body: payload,
+        }
+    );
+
+    if (!res.ok) {
+        const error = await res.json();
+        throw {
```

Comment:
Not sure if we should throw an error in this function, maybe it would be better to just return false or something, but we might as well be consistent with the rest of the code

However if !res.ok we should also log the error to sentry with logException(..) (see other functions)

Also we should be throwing an error object created with Javascript's Error() constructor, not sure what will happen if you pass it this data shape instead